src/twisted/protocols/postfix.py

```diff
@@ -105,7 +107,13 @@ def do_put(self, keyAndValue):
                 self.sendCode(500, b"put is not implemented yet.")
 
 
-class PostfixTCPMapDictServerFactory(UserDict, protocol.ServerFactory):
+if TYPE_CHECKING or sys.version_info >= (3, 9):
+    _PostfixTCPMapDict = UserDict[bytes, Union[str, bytes]]
+else:
+    _PostfixTCPMapDict = UserDict
+
+
+class PostfixTCPMapDictServerFactory(_PostfixTCPMapDict, protocol.ServerFactory):
```

Comment:
a bit horrible, maybe this should be `typing.Dict` instead of `collections.UserDict` one day

Reply:
it should be ok for this PR